sklearn/neighbors/base.py

```diff
@@ -258,6 +258,19 @@ def _fit(self, X):
         else:
             raise ValueError("algorithm = '%s' not recognized"
                              % self.algorithm)
+
+        if self.n_neighbors is not None:
```

Comment:
It is set to ``None`` in ``_init_params``, right? Don't you think this is the right fix?